### PR TITLE
Support did:key resolution in 2018/2019 crypto suite mode.

### PR DESCRIPTION
**Pull request by dmitrizagidulin**
Also, no longer bundling the various contexts with this lib (reduce bundle size).

### CODE REVIEW

File: lib/main.js [hunk 1]
```diff
@@ -10,14 +10,11 @@ import {DidKeyDriver} from './DidKeyDriver.js';
...
- * @param {Function} [options.convert] - An async function to convert a
- *   verification key pair (like ed25519) to a key-agreement key pair (such
- *   as x25519).
  *
  * @returns {DidKeyDriver} Returns an instance of a did:key resolver driver.
  */
-function driver({verificationSuite, convert} = {}) {
-  return new DidKeyDriver({verificationSuite, convert});
+function driver({verificationSuite} = {}) {
+  return new DidKeyDriver({verificationSuite});
```
Comment (mattcollier) [inline lib/main.js:17]: This looks like a breaking change to me, you're removing an optional parameter.  Existing code written to this API would no longer function.  Changelog indicates this is slated to be a minor release.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,11 @@
 # did:key driver ChangeLog
 
+## 1.2.0 -
```
Comment (mattcollier) [inline CHANGELOG.md:3]: See comments below about release type due to changes in `driver` API.

**Review by mattcollier — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -254,6 +254,44 @@ const {verify} = assertionMethodPublicKey.verifier();
...
+### Backwards Compatibility with the 2018/2019 Crypto Suites
+
+By default, this did key driver returns DID Documents that have the 2020
+crypto suites for verification and key agreement.
+If you need DID Documents that are using the 2018/2019 crypto suites,
+you can customize the driver as follows.
+
+```js
+const didKeyDriver2018 = driver({
+ verificationSuite: Ed25519VerificationKey2018
```
Comment (davidlehn) [inline README.md:266]: Should that value be a string or a var imported from elsewhere?

File: lib/main.js (hunk 1 above)
Comment (dmitrizagidulin) [reply lib/main.js:17]: @mattcollier right, that makes sense. This was just a cleanup though - the constructor doesn't actually have that parameter defined. (Meaning, this `driver()` function was passing it in, but nothing is receiving on the other end.) That's why I was thinking non breaking.

File: lib/DidKeyDriver.js [hunk 4]
```diff
@@ -242,18 +267,18 @@ export function _getKey({didDocument, keyIdFragment}) {
...
+    // Return the public key node for the X25519 key-agreement key
+    publicKey = didDocument.keyAgreement[0];
   }
-  // Return the public key node for the X25519 key-agreement key
+
   return {
-    '@context': x25519Context.constants.CONTEXT_URL,
-    ...didDocument.keyAgreement[0]
+    '@context': contextsBySuite.get(publicKey.type),
+    ...publicKey
```
[the remark this answers is not in the document]
Comment (dmitrizagidulin) [reply lib/DidKeyDriver.js:282]: There is, yeah - for `LDKeyPair` instances. This is just a plain json object, hence why I have to do it manually.

File: lib/DidKeyDriver.js [hunk 5]
```diff
@@ -176,10 +188,27 @@ export class DidKeyDriver {
...
     // The KAK pair will use the source key's controller, but will generate
     // its own .id
-    const keyAgreementKeyPair = X25519KeyAgreementKey2020
-      .fromEd25519VerificationKey2020({keyPair: verificationKeyPair});
+    let keyAgreementKeyPair;
+    if(verificationKeyPair.type === 'Ed25519VerificationKey2020') {
+      keyAgreementKeyPair = X25519KeyAgreementKey2020
+        .fromEd25519VerificationKey2020({keyPair: verificationKeyPair});
+      contexts.push(Ed25519VerificationKey2020.SUITE_CONTEXT);
+      contexts.push(X25519KeyAgreementKey2020.SUITE_CONTEXT);
```
[the remark this answers is not in the document]
Comment (dmitrizagidulin) [reply lib/DidKeyDriver.js:200]: @mattcollier - oh cool! did not know that.

File: README.md (hunk 3 above)
Comment (dmitrizagidulin) [reply README.md:266]: Ah, yeah, I'll add the import.

File: lib/main.js (hunk 1 above)
Comment (mattcollier) [reply lib/main.js:17]: OK, thanks for clarification.

File: README.md (hunk 3 above)
Comment (dmitrizagidulin) [reply README.md:271]: Added

File: lib/DidKeyDriver.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (dmitrizagidulin) [reply lib/DidKeyDriver.js]: Adjusted, thanks.

File: README.md [hunk 6]
```diff
@@ -254,6 +254,44 @@ const {verify} = assertionMethodPublicKey.verifier();
 `publicMethodFor` will return `undefined` if no key is found for a given
 purpose.
 
+### Backwards Compatibility with the 2018/2019 Crypto Suites
+
+By default, this did key driver returns DID Documents that have the 2020
```
[the remark this answers is not in the document]
Comment (dmitrizagidulin) [reply README.md]: Thanks (went with `did:key`).

**Pull request merged**